core/src/scene/styleParam.cpp

```diff
@@ -592,106 +545,125 @@ bool StyleParam::parseTime(const std::string &_value, float &_time) {
 }
 
 template<typename T>
-int parseVec(const std::string& _value, T& _vec) {
-
-    const size_t elements = _vec.length();
-    const char* str = _value.data();
-
-    const int length = _value.length();
-    int count = 0;
-    int offset = 0;
-
-    for (size_t i = 0; i < elements; i++) {
-
-        float v = ff::stof(str + offset, length - offset, &count);
-        if (count == 0) { return i; }
-
-        _vec[i] = v;
-        offset += count;
-
-        if (length - offset <= 0) { return i; }
+bool parseVec(const YAML::Node& node, UnitSet allowedUnits, UnitVec<T>& vec) {
+    if (!node.IsSequence()) {
+        return false;
+    }
+    size_t nodeSize = node.size();
+    bool success = true;
+    for (size_t i = 0; i < vec.size; i++) {
+        if (i >= nodeSize) {
+            success = false;
+            break;
+        }
+        const auto& nodeElement = node[i];
+        if (!nodeElement.IsScalar()) {
+            success = false;
+            continue;
+        }
+        StyleParam::ValueUnitPair result;
+        success |= StyleParam::parseValueUnitPair(nodeElement.Scalar(), result);
 
-        // Skip next comma
-        while (str[offset] == ' ') { offset++; }
-        if (str[offset++] != ',') { return i; }
+        if (!allowedUnits.contains(result.unit)) {
+            success = false;
+        }
+        vec.units[i] = result.unit;
+        vec.value[i] = result.value;
     }
 
-    return elements;
+    return success;
 }
 
-template<typename T>
-int parseVec(const std::string& _value, uint8_t allowedUnits, UnitVec<T>& _vec) {
-     // initialize with defaults
-    const size_t elements = _vec.size;
-    for (size_t i = 0; i < elements; i++) {
-        _vec.units[i] = Unit::pixel;
-        _vec.value[i] = NAN;
+bool StyleParam::parseSize(const YAML::Node& node, UnitSet allowedUnits, SizeValue& result) {
+    bool success = false;
+    if (node.IsScalar()) {
+        success = parseSizeUnitPair(node.Scalar(), result.x);
     }
-
-    int offset = 0;
-    for (size_t i = 0; i < elements; i++) {
-        StyleParam::ValueUnitPair v;
-        offset = StyleParam::parseValueUnitPair(_value, offset, v);
-        if (offset < 0) { return i; }
-
-        if ( !(v.unit & allowedUnits) ) {
-            return 0;
-        }
-        _vec.units[i] = v.unit;
-        _vec.value[i] = v.value;
+    if (node.IsSequence() && node.size() >= 2) {
```

Comment:
it looks like you have it parsing for just the first 2 units if size is a sequence. Maybe it makes sense to also warn if size sequence is greater than 2 and still allow the 1st 2 values to be parsed for size.

Reply:
Sure, could warn if there are more than 2 elements in the sequence.

Reply:
On second thought, I'd rather err on the side of simplicity here and not warn for something harmless like this.